src/grid_interpolator.cc

```diff
@@ -0,0 +1,82 @@
+#include "flyft/grid_interpolator.h"
+
+#include <cmath>
+#include <cassert>
+#include <fstream>
+#include <tuple> 
+
+namespace flyft
+{
+GridInterpolator::GridInterpolator(const std::string& s)
+    {
+    std::ifstream indata(s);
+    if(!indata)
+        {
+        throw std::runtime_error("File failed to open");
+        }
+        
+    int nx, ny, nz;   
+    indata >> nx >> ny >> nz >> dx_ >> dy_ >> dz_;
+    n_ = ThreeDimensionIndex(nx,ny,nz);
+    
+    if (n_.size() == 0)
+        {
+        throw std::runtime_error("Grid has zero size");
+        }
+        
+    data_ = new double[n_.size()];  
+    double value;
+    
+    int idx = 0;
+    while (indata >> value)
+        {
+        data_[idx++] = value;
+        }
+        
+    if (idx != n_.size() || !indata.eof())
+        {
+        throw std::runtime_error("File did not right number of elements");
+        }
+        
+    indata.close();
+    }
+
+GridInterpolator::~GridInterpolator()
+    {
+    delete[] data_;
+    }
+
+double GridInterpolator::operator()(double x, double y, double z) const
+    {
+    const int bin_x = std::floor(x/dx_); 
+    const int bin_y = std::floor(y/dy_); 
+    const int bin_z = std::floor(z/dz_); 
+    
+    assert(bin_x >= 0 && bin_x < std::get<0>(n_.getBounds()));
```

Comment:
Do still store the bounds once, it is more readable. You just don't need to unpack it into 3 vars. Like so:
```suggestion
    const auto bounds = n_.getBounds();
    assert(bin_x >= 0 && bin_x < std::get<0>(bounds)));
```
apply to the next two lines as well.